src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/AbstractHostObject.cs

```diff
@@ -7,12 +7,22 @@
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
 {
+
     /// <summary>
     /// Base type for language service host object for cross targeting projects.
     /// </summary>
-    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider
+    internal abstract class AbstractHostObject : IVsHierarchy, IVsContainedLanguageProjectNameProvider, IVsProject
     {
-        protected abstract IVsHierarchy InnerHierarchy { get; }
+        protected AbstractHostObject(object innerHostObject)
```

Comment:
In that case, can we just add a doc comment here, saying what we expect the object to implement? Every time we lose type-safety I cringe, but let's at least document it.